extensions/iceberg/src/test/java/io/deephaven/iceberg/junit5/SqliteCatalogBase.java

```diff
@@ -1039,6 +1048,399 @@ void testAutoRefreshingPartitionedAppend() throws InterruptedException {
         assertTableEquals(expected2, fromIcebergRefreshing.select());
     }
 
+    /**
+     * Verify that the sort order for the data files in the table match the expected sort order.
+     */
+    private void verifySortOrder(
+            final IcebergTableAdapter tableAdapter,
+            final List<List<SortColumn>> expectedSortOrders) {
+        verifySortOrder(tableAdapter, expectedSortOrders,
+                ParquetInstructions.EMPTY.withTableDefinition(tableAdapter.definition()));
+    }
+
+    private void verifySortOrder(
+            @NotNull final IcebergTableAdapter tableAdapter,
+            @NotNull final List<List<SortColumn>> expectedSortOrders,
+            @NotNull final ParquetInstructions readInstructions) {
+        final org.apache.iceberg.Table icebergTable = tableAdapter.icebergTable();
+        final Map<ManifestFile, List<DataFile>> manifestToDataFiles =
+                IcebergUtils.manifestToDataFiles(icebergTable, icebergTable.currentSnapshot());
+        final List<List<SortColumn>> actualSortOrders = new ArrayList<>();
+        for (final Map.Entry<ManifestFile, List<DataFile>> entry : manifestToDataFiles.entrySet()) {
+            final List<DataFile> dataFiles = entry.getValue();
+            for (final DataFile dataFile : dataFiles) {
+                actualSortOrders.add(computeSortedColumns(icebergTable, dataFile, readInstructions));
+            }
+        }
+        assertThat(actualSortOrders).containsExactlyInAnyOrderElementsOf(expectedSortOrders);
```

Comment:
Does this not work when you use `isEqualTo`? We should be get the data in the order it was added if it is not already.